pkg/container-collection/containers.go

```diff
@@ -21,49 +21,49 @@ import (
 // Container represents a container with its metadata.
 type Container struct {
 	// ID is the container id, typically a 64 hexadecimal string
-	ID string
+	ID string `json:"id,omitempty"`
 
 	// Pid is the process id of the container
-	Pid uint32
+	Pid uint32 `json:"pid,omitempty"`
 
 	// Container's configuration is the config.json from the OCI runtime
 	// spec
-	OciConfig *ocispec.Spec
+	OciConfig *ocispec.Spec `json:"oci_config,omitempty"`
 
 	// Bundle is the directory containing the config.json from the OCI
 	// runtime spec
 	// See https://github.com/opencontainers/runtime-spec/blob/main/bundle.md
-	Bundle string
+	Bundle string `json:"bundle,omitempty"`
 
 	// Linux metadata can be derived from the pid via /proc/$pid/...
-	Mntns      uint64
-	Netns      uint64
-	CgroupPath string
-	CgroupID   uint64
+	Mntns      uint64 `json:"mntns,omitempty"`
+	Netns      uint64 `json:"netns,omitempty"`
+	CgroupPath string `json:"cgroup_path,omitempty"`
+	CgroupID   uint64 `json:"cgroup_id,omitempty"`
 	// Data required to find the container to Pod association in the
 	// gadgettracermanager.
-	CgroupV1 string
-	CgroupV2 string
+	CgroupV1 string `json:"cgroupv1,omitempty"`
+	CgroupV2 string `json:"cgroupv2,omitempty"`
 
 	// Kubernetes metadata
-	Namespace string
-	Podname   string
-	Name      string
-	Labels    map[string]string
+	Namespace string            `json:"namespace,omitempty"`
+	Podname   string            `json:"podname,omitempty"`
+	Name      string            `json:"name,omitempty"`
+	Labels    map[string]string `json:"labels,omitempty"`
 	// The owner reference information is added to the seccomp profile as
 	// annotations to help users to identify the workflow of the profile.
-	OwnerReference *OwnerReference
-	PodUID         string
+	OwnerReference *OwnerReference `json:"owner_reference,omitempty"`
+	PodUID         string          `json:"pod_uid,omitempty"`
 
 	// Container Runtime metadata
-	Runtime string
+	Runtime string `json:"runtime,omitempty"`
 }
 
 type OwnerReference struct {
-	Apiversion string
-	Kind       string
-	Name       string
-	UID        string
+	Apiversion string `json:"api_version,omitempty"`
+	Kind       string `json:"kind,omitempty"`
+	Name       string `json:"name,omitempty"`
+	UID        string `json:"uid,omitempty"`
```

Comment:
I agree we should switch to the Kubernetes types instead of redefining `OwnerReference`.

On the other hand, even though the users of local-gadget will not see the `OwnerReference` because such information cannot be retrieved from the container runtime (at least, that's what I know), I would prefer to use `camelCase` for the `type Container` as the `type OwnerReference` from already Kubernetes uses that notation. WDYT?